sdk/rust/core/src/token.rs

```diff
@@ -216,6 +263,28 @@ pub enum Action {
     /// Upgrades the token bridge contract to a new address.
     #[serde(rename = "2")]
     ContractUpgrade { new_contract: Address },
+
+    // Modify balance for tokenbridge
+    #[serde(rename = "3")]
+    ModifyBalance {
+        // module:  \x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00TokenBridge (32)
+        // chainId: \x03\x0c (2)
+        // action: \x03 (1)
+
+        // \x00\x00\x00\x00\x00\x00\x00$
+        sequence: u64, // 8
+
+        // \x00\x01
+        chain_id: u16, // 10
+        // \x00\x01
+        token_chain: u16, // 12
+        // ||||||||||||||||||||||||||||||||
+        token_address: Address, // 44
+        // \x01
+        kind: ModificationKind, // 45
+        amount: Amount,         // 78
+        reason: [u8; 32],       // 110
```

Comment:
We have the `arraystring` module for this (see also `Message::AssetMeta`):

```suggestion
        #[serde(with = "crate::arraystring")]
        reason: BString,       // 110
```